docs/developer-guides/migration-guide-1.4.0.md

```diff
@@ -68,12 +69,21 @@ Because of the changes to the smart contracts and their on-chain data, a complet
 of main chain data needs to be set up on Cardano.
 
 1. Pick the initial governance authority.
-This can be a newly generated or already existing main chain key set controlled by the Partner Chain authority.
-2. Run the `prepare-configuration` command of `partner-chains-cli` (v1.4.0) in a fresh directory.
+This can be a newly generated or already existing main chain key set (including the current governance authority of the Partner Chain being upgraded)
+controlled by the Partner Chain authority.
+The address associated with the keys should have enough ADA to cover transaction costs (10 or more ADA is advised).
+2. Run the `generate-keys` command of `partner-chains-cli` (v1.4.0). The generated keys and keystore are
+required by other commands and can be discarded when finished.
```

Comment:
This will be a rabbit hole. If chain is upgraded and governance authority is not running node then maybe it needs to run this command. Otherwise it should run `prepare-configuration` in the same directory as the last time.

Reply:
I replaced this point with a notice explaining the two possibilities.